src/Tooltip.js

```diff
@@ -50,23 +58,25 @@ class Tooltip extends React.Component {
   }
 
   onMouseOverTooltip() {
-    if (this._hoverTimeout) {
-      clearTimeout(this._hoverTimeout);
-    }
-
-    if (!this.props.isOpen) {
-      this.toggle();
+    if (this._hideTimeout) {
+      clearTimeout(this._hideTimeout);
     }
+    this._showTimeout = setTimeout(this.show, this.getDelay('show'));
   }
 
   onMouseLeaveTooltip() {
-    this._hoverTimeout = setTimeout(this.onTimeout, 250);
+    if (this._showTimeout) {
+      clearTimeout(this._showTimeout);
+    }
+    this._hideTimeout = setTimeout(this.hide, this.getDelay('hide'));
   }
 
-  onTimeout() {
-    if (this.props.isOpen) {
-      this.toggle();
+  getDelay(key) {
```

Comment:
Instead of running this on every mouse over and mouse leave, why not do this when it mounts and each time props update. It will help avoid overhead.